make find-references behave better with simpler code

kermorgant opened the pull request:
ensime-mode has a similar implementation, which does not rely on
tabulated mode and I liked the shorter implementation (+ grep-mode makes
it easy to kill the buffer with a single keystroke).

Besides, set-window-dedicated-p helps make the buffer sticky.

partly solves #98

kermorgant commented:
I might give it another stab current appearance of ref list is not as good as with tabulated mode

zonuexe commented on phpactor.el:
```diff
@@ -356,38 +356,27 @@ of GitHub.")
+    (erase-buffer)
+    (dolist (file-reference current-references)
+      (let ((path (plist-get file-reference :file))
+            (x nil))
```
Is `x` unused?

zonuexe commented on phpactor.el:
```diff
@@ -356,38 +356,27 @@ of GitHub.")
+          (when path
+            (insert-text-button (phpactor-truncate-left path phpactor-references-list-col1-width)
+                        'action (lambda(x) (find-file path (plist-get reference :start)))
+                        'help-echo "mouse-2: visit this file in other window")
```
* If you ignore the argument, the variable name should be `_`.
* Please indent this line up properly.
* Separate between `lambda` and the argument list with a space.

zonuexe commented:
I commented on details, but it looks good to me.

kermorgant commented:
@zonuexe 
I added goto-char instruction, reordered the "cols" (more like the initial tabulated mode) and fixed the things you mentioned (except maybe the indent ?)
